### DISCUSSION TITLE
Bprime command in case of chlorine based molecules

### DISCUSSION THREAD

**Discussion by RaminShilav** [mutationpp/Mutationpp#168] [Q&A] [answered]
hi all,
I am trying to generate B' tables for a simulation of TPS ablation inside a solid rocket motor. All was well until I added chlorine-based molecules (among other C-H-O molecules) to the mixture file in the "free stream".
The output seems unreasonable, with chlorine alone. 
Does anyone know what am I doing wrong ?

Best Regards,
Ramin Shilav, RAFAEL LTD

**Comment by grgbellasvki** [comment]
Dear Shilav, 
Would you mind sharing your xml file here so we can review it. Did you try checking it with "checkmix" command to ensure it is ok?
Thanks,
George

**Comment by RaminShilav** [comment]
Hi george
Thank you for your reply.
Checkmix was ok.
Until now, I managed to get reasonsble results by adding  halogen element
"contaminant" to the mixture in a very small fraction. Still, I will be
happy to know if there is a more appropriate way.

Best regards
Ramin Shilav

בתאריך יום ב׳, 12 ביולי 2021, 20:38, מאת grgbellasvki ‏<
***@***.***>:

**Comment by RaminShilav** [comment]
attached the content of xml file ( I omitted some signs of command start and ending):

mixture mechanism="none" thermo_db="NASA-9"

species
 C H O N CL CH4  CO CO2 C2 HCL C2H2,acetylene HCN H2 H2O N2 OH HNC C(gr)
species

element_compositions default="free-stream"
composition name="pyrolysis"
 C  : 0.06 , H  : 0.889   , O  : 0.05  , N  : 0.01  
composition

composition name="free-stream"
C: 0.1   ,   H: 0.5     ,   O: 0.2    ,  N: 0.15 , CL: 0.05
composition
element_compositions
mixture

Ramin

**Comment by grgbellasvki** [comment, marked as answer]
Dear Ramin,

Glad to hear you are getting reasonable results. We are always happy to see in the forum verification of the library with other literature results for further discussion.

I am not an expert in chlorine chemistry, but you could try the following mixture, which includes all possible gaseous species in the NASA-9 database composed of C, H, N, O or Cl. The equilibrium solver will physically discard the thermodynamically unstable ones.

```
<mixture mechanism="none" thermo_db="NASA-9">

    <species>
        { gases with C, H, O, N, Cl } C(gr)
    </species>

    <element_compositions default="freestream">

        <composition name="pyrolysis"> 
	    C:0.06, H:0.889, O:0.05 , N:0.01, Cl:0.
        </composition>

        <composition name="freestream"> 
	    C:0.1, H:0.5, O:0.2, N:0.15, Cl:0.05
        </composition>

    </element_compositions>

</mixture>
```

George

**Reply by RaminShilav** [reply]
Thank you. I will try this option.
Ramin
